Upgrade @firebase/database (and possibly other @firebase/ dependencies)

https://github.com/firebase/firebase-admin-node/blob/be4ebc61ed013315ffad954575f6223d44b5a80a/package.json#L59

Currently the `@firebase/database` dependency is limited to `^0.8.1` but there have been a [few releases](https://github.com/firebase/firebase-js-sdk/blob/master/packages/database/CHANGELOG.md) since then - mainly the new `Query#startAfter` and `Query#endBefore` methods added in `0.9.X`.

I've tried upgrading `yarn add firebase-admin@latest` and also `yarn upgrade` and not getting a later `@firebase/database` version than `0.8.3` - but that does make sense as `^0.8.1` will only allow patch updates on the `0.8.X` line...

I believe this may also be the root cause of the issue we are seeing in our app (see https://github.com/sparkletown/sparkle/pull/1362#issuecomment-833438685), trying to use the latest `firebase` (`8.5.0`) with the latest `firebase-admin` (`8.7.0`).

See the linked comment for more specific details, but at a high level, it seems as though `firebase@8.5.0` requires  a `1.x` version of `@firebase/util` and `firebase-admin@8.7.0` requires a `0.x` version of `@firebase/util`. So having them together in the same package leads to issues.

---

This leads to the following warnings when installing the packages:

By default:
```
npm WARN @firebase/auth-types@0.10.3 requires a peer of @firebase/util@1.x but none is installed. You must install peer dependencies yourself.
npm WARN @firebase/firestore-types@2.3.0 requires a peer of @firebase/util@1.x but none is installed. You must install peer dependencies yourself.
npm WARN @firebase/storage-types@0.4.1 requires a peer of @firebase/util@1.x but none is installed. You must install peer dependencies yourself.
```

If I manually install `npm i @firebase/util@1.1.0`:

```
npm WARN @firebase/auth-interop-types@0.1.5 requires a peer of @firebase/util@0.x but none is installed. You must install peer dependencies yourself.
```

We can see the issue with `npm ls @firebase/util`:

```
⇒  npm ls @firebase/util
co-reality-map@0.1.0 /Users/devalias/dev/sparkle/sparkle
├── @firebase/util@1.1.0
├─┬ firebase@8.5.0
│ ├─┬ @firebase/analytics@0.6.10
│ │ ├─┬ @firebase/component@0.5.0
│ │ │ └── @firebase/util@1.1.0  deduped
│ │ └── @firebase/util@1.1.0
│ ├─┬ @firebase/app@0.6.21
│ │ ├─┬ @firebase/component@0.5.0
│ │ │ └── @firebase/util@1.1.0  deduped
│ │ └── @firebase/util@1.1.0
│ ├─┬ @firebase/database@0.10.0
│ │ ├─┬ @firebase/component@0.5.0
│ │ │ └── @firebase/util@1.1.0  deduped
│ │ └── @firebase/util@1.1.0  deduped
│ ├─┬ @firebase/firestore@2.3.0
│ │ ├─┬ @firebase/component@0.5.0
│ │ │ └── @firebase/util@1.1.0  deduped
│ │ └── @firebase/util@1.1.0
│ ├─┬ @firebase/functions@0.6.8
│ │ └─┬ @firebase/component@0.5.0
│ │   └── @firebase/util@1.1.0
│ ├─┬ @firebase/installations@0.4.26
│ │ ├─┬ @firebase/component@0.5.0
│ │ │ └── @firebase/util@1.1.0  deduped
│ │ └── @firebase/util@1.1.0
│ ├─┬ @firebase/messaging@0.7.10
│ │ ├─┬ @firebase/component@0.5.0
│ │ │ └── @firebase/util@1.1.0  deduped
│ │ └── @firebase/util@1.1.0
│ ├─┬ @firebase/performance@0.4.12
│ │ ├─┬ @firebase/component@0.5.0
│ │ │ └── @firebase/util@1.1.0  deduped
│ │ └── @firebase/util@1.1.0
│ ├─┬ @firebase/remote-config@0.1.37
│ │ ├─┬ @firebase/component@0.5.0
│ │ │ └── @firebase/util@1.1.0  deduped
│ │ └── @firebase/util@1.1.0
│ ├─┬ @firebase/storage@0.5.2
│ │ ├─┬ @firebase/component@0.5.0
│ │ │ └── @firebase/util@1.1.0  deduped
│ │ └── @firebase/util@1.1.0
│ └── @firebase/util@1.1.0
└─┬ firebase-admin@9.7.0
  └─┬ @firebase/database@0.8.3
    ├─┬ @firebase/component@0.1.21
    │ └── @firebase/util@0.3.4
    └── UNMET PEER DEPENDENCY @firebase/util@0.3.4

npm ERR! peer dep missing: @firebase/util@0.x, required by @firebase/auth-interop-types@0.1.5
```

--- a/package.json
+++ b/package.json
@@ -56,8 +56,8 @@
   ],
   "types": "./lib/index.d.ts",
   "dependencies": {
-    "@firebase/database": "^0.8.1",
-    "@firebase/database-types": "^0.6.1",
+    "@firebase/database": "^0.10.0",
+    "@firebase/database-types": "^0.7.2",
     "@types/node": "^10.10.0",
     "dicer": "^0.3.0",
     "jsonwebtoken": "^8.5.1",
@@ -68,9 +68,9 @@
     "@google-cloud/storage": "^5.3.0"
   },
   "devDependencies": {
-    "@firebase/app": "^0.6.13",
-    "@firebase/auth": "^0.16.2",
-    "@firebase/auth-types": "^0.10.1",
+    "@firebase/app": "^0.6.21",
+    "@firebase/auth": "^0.16.5",
+    "@firebase/auth-types": "^0.10.3",
     "@microsoft/api-extractor": "^7.11.2",
     "@types/bcrypt": "^2.0.0",
     "@types/chai": "^4.0.0",

